android-json-form-wizard/src/main/java/com/vijay/jsonwizard/task/AttachRepeatingGroupTask.java

```diff
@@ -150,7 +155,24 @@ private LinearLayout buildRepeatingGroupLayout(ViewParent parent) throws Excepti
                 }
                 // add element to json form object to be written into
                 JSONObject step = ((JsonApi) widgetArgs.getContext()).getmJSONObject().getJSONObject(widgetArgs.getStepName());
-                step.getJSONArray(FIELDS).put(element);
+                JSONArray fields = step.getJSONArray(FIELDS);
+                int repeatingGroupIndex = 0;
+                JSONArray updatedFields = new JSONArray();
+                String repeatingGroupKey = widgetArgs.getJsonObject().getString(KEY);
+                for(int fieldIndex = 0; fieldIndex < fields.length(); fieldIndex++) {
+                    JSONObject fieldObject = fields.getJSONObject(fieldIndex);
+                    updatedFields.put(fieldObject);
+                    if(fieldObject.has(KEY) && fieldObject.getString(KEY).equals(repeatingGroupKey)) {
+                        repeatingGroupIndex = fieldIndex;
+                        break;
+                    }
+                }
+                updatedFields.put(element);
+                for(int fieldIndex = repeatingGroupIndex + 1; fieldIndex < fields.length(); fieldIndex++) {
+                    JSONObject fieldObject = fields.getJSONObject(fieldIndex);
+                    updatedFields.put(fieldObject);
+                }
+                step.put(FIELDS, updatedFields);
```

Comment:
It's not very clear what is happening here, could you please add comments that explain the algorithm?

Reply:
I have added in-line comments explaining the logic.

Reply:
Cool, more understandable now, thanks.

I think we should store the `repeatingGroupIndex` as a global variable, initially equal to `-1` then after it's initialized to it's true value, we won't need to do this traversal again (which is expensive).

So we can factor out this logic to a separate function and only call the function if `repeatingGroupIndex == 1`?

Reply:
Re-initialization to `-1` would happen in the `getViewsFromJson` function.

Reply:
Could the logic move to a separate private function with a descriptive name for what is happening?

The helpful docstring you added can then be put above the function signature.